Source/ARTHttp.m

```diff
@@ -12,21 +12,34 @@
 
 @interface ARTHttp ()
 
-@property (readonly, strong, nonatomic) ARTURLSessionServerTrust *urlSession;
+@property (readonly, strong, nonatomic) id<ARTURLSession> urlSession;
 
 @end
 
+Class urlSessionClass = nil;
+
 #pragma mark - ARTHttp
 
 @implementation ARTHttp {
     ARTLog *_logger;
     _Nullable dispatch_queue_t _queue;
 }
 
++ (void)setURLSessionClass:(Class)class {
+    urlSessionClass = class;
+}
+
 - (instancetype)init:(dispatch_queue_t)queue logger:(ARTLog *)logger {
+    static dispatch_once_t setURLSessionClassOnce;
+    dispatch_once(&setURLSessionClassOnce, ^{
```

Comment:
This feels a little over engineered... `[ARTURLSessionServerTrust class]` will always be the same (static by nature) so at the point of assignment of the `_urlSession` ivar, couldn't you snapshot `urlSessionClass` at that point? e.g.:

```
const Class configuredClass = urlSessionClass;
const Class class = configuredClass ? configuredClass : [ARTURLSessionServerTrust class];
_urlSession = [[class alloc] init:queue];
```

Reply:
Yes, that looks better, thanks. Done at 302d146.